hardware/Netatmo.cpp

```diff
@@ -504,12 +576,445 @@ uint64_t CNetatmo::UpdateValueInt(int HardwareID, const char* ID, unsigned char
+							oldDevType,
+							oldDevSubType,
+							oldUnit,
+							oldUsed,
+							oldLastUpdate.c_str()
+						);
+						MergeDevices(oldID, deviceRowIdx);
+						break; 	//One device per cycle
+					}
+					else if (
+						unit == 140
+						&& oldUnit == 0
+						&& oldDevType == devType
+						&& oldDevType == pTypeTEMP_HUM_BARO	// (from RFXtrx.h)
+						&& oldDevSubType == sTypeTHBFloat
+						&& oldDevSubType == devSubType
+						&& m_iNetatmoProtVersion <= 2
+						//&& oldLastUpdate < lastUpdate
+					) 
+					{
+						Log(LOG_STATUS, "case 9 (Back to old  unit number for weather station): Device: ID=%" PRIu64 ", DeviceID=0x%s), Name=%s, Type=%d, devSubType=%d, unit=%d, used=%d, lastUpdate=%s",
+							oldID,
+							oldDeviceID.c_str(),
+							device[2].c_str(),
+							oldDevType,
+							oldDevSubType,
+							oldUnit,
+							oldUsed,
+							oldLastUpdate.c_str()
+						);
+						MergeDevices(oldID, deviceRowIdx);
+						break; 	//One device per cycle
+					}
+				}
+			}
+		}
+	}
+}
+
+
+/// <summery>
+/// Merge old and new devices, bu deleting duplicate log records and
+/// copy all other log records from the new to the old device before
+/// updating the old device to the new devices and finaly remove the new device.
+/// Note: the old device will become the new device.
+/// </summary>
+bool CNetatmo::MergeDevices(const uint64_t ipOldDeviceId, const uint64_t ipNewDeviceId)
+{
+//	Log(LOG_STATUS, "MergeDevices: Merge old device %" PRIu64 " with new device %" PRIu64 ".", ipOldDeviceId, ipNewDeviceId);
+
+	m_iMigrationsDone++;
+
+	std::string sOldDeviceId = std::to_string(ipOldDeviceId);
+	std::string sNewDeviceId = std::to_string(ipNewDeviceId);
+
+	MergeDeviceLogs("Fan", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("Fan_Calendar", sOldDeviceId, sNewDeviceId);
+
+	MergeDeviceLogs("Meter", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("Meter_Calendar", sOldDeviceId, sNewDeviceId);
+
+	MergeDeviceLogs("MultiMeter", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("MultiMeter_Calendar", sOldDeviceId, sNewDeviceId);
+
+	MergeDeviceLogs("Percentage", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("Percentage_Calendar", sOldDeviceId, sNewDeviceId);
+
+	MergeDeviceLogs("Rain", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("Rain_Calendar", sOldDeviceId, sNewDeviceId);
+
+	MergeDeviceLogs("Temperature", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("Temperature_Calendar", sOldDeviceId, sNewDeviceId);
+
+	MergeDeviceLogs("UV", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("UV_Calendar", sOldDeviceId, sNewDeviceId);
+
+	MergeDeviceLogs("Wind", sOldDeviceId, sNewDeviceId);
+	MergeDeviceLogs("Wind_Calendar", sOldDeviceId, sNewDeviceId);
+
+	return m_sql.TransferDevice(sOldDeviceId, sNewDeviceId);
+	 Log(LOG_STATUS, "Merge completed");
+}
+
+
+/// <summary>
+/// Merge logs in all log tables
+/// </summary>
+bool CNetatmo::MergeDeviceLogs (const std::string& spTableName, const std::string& spOldDeviceId, const std::string& spNewDeviceId)
```

Comment:
Could you remove the space behind the function name, or in visual studio press control-k followed by normal 'd'

Reply:
Done